tools/sof-ipc-timer.py

```diff
@@ -0,0 +1,808 @@
+#!/usr/bin/env python3
+
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright(c) 2025 Intel Corporation. All rights reserved.
+# pylint: disable=line-too-long, too-few-public-methods, missing-function-docstring
+# pylint: disable=too-many-arguments, consider-using-f-string, too-many-instance-attributes
+# pylint: disable=too-many-locals, too-many-return-statements, too-many-format-args, invalid-name
+# pylint: disable=too-many-public-methods, consider-using-dict-items, consider-using-with
```

Comment:
Many of these `disable` should be on a per-function or even per-line basis. Having a few functions with too many arguments is not an issue (I see only one right now?), but all of them having too many arguments is not. "consider-using-with" is a very valuable warning and advice, silencing it for the **entire** script does not seem like a good idea.